src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml

```diff
@@ -8,6 +8,7 @@
     </type>
     <type fullname="System.Globalization.GlobalizationMode">
       <method signature="System.Boolean get_Invariant()" body="stub" value="true" feature="System.Globalization.Invariant" featurevalue="true" />
+      <method signature="System.Boolean get_Invariant()" body="stub" value="false" feature="System.Globalization.Invariant" featurevalue="false" />
```

Comment:
Are there other cases like this?  I'm wondering whether we need more cases of handling both the true and false values, and if so, if there's a more general feature needed here, where the linker sees a line like:
```C#
<method signature="System.Boolean get_Invariant()" body="stub" feature="System.Globalization.Invariant" featurevalue="true" />
```
and automatically propagates `featurevalue` to `value`, or something like that.

Reply:
I had proposed this in chat to @eerhardt the other day since it means we are potentially leaving behind bits of "unused" code that are statically known to only be `true`/`false` and therefore have no side effects.

https://github.com/mono/linker/issues/1106#issuecomment-774247286 is a related example where just leaving the `get_IsSupported()` methoods can lead to larger downstream impact due to inheritance chains.

Reply:
In all our existing feature switches, only 1 value will trim any substantial piece of code. For Invariant, I wasn't aware how much unnecessary code was still being rooted for the default setting. I thought only `Invariant=true` would trim a decent amount of code.

But I can imagine this being beneficial in other feature switches in the future.

cc @marek-safar @vitek-karas @sbomer - thoughts?

Reply:
A related bit is that even needing to hardcode `get_Invariant` and others seems potentially unnecessary in certain cases

One would expect the linker could recognize the general pattern of `static readonly bool s_name = AppContext.TryGetSwitch("", out bool result) ? result : default` (or others) and that we wouldn't need to hardcode a table of substitutions outside of more complicated scenarios.
It, plus knowing that `static readonly <primitive> name = value` can't change after being set seem like generally good things to recognize as they can lead to later downstream optimizations and therefore trimming opportunities

Reply:
If it's common we could make syntax which would be able to substitute the value directly from the feature value. Note that even boolean features have 3 values: `true`, `false`, `not-set`. I assume the "not-set" would mean no substitution (unless the `defaultfeaturevalue` is used).

If this is the only place right now, it's probably not worth it though.

Reply:
> One would expect the linker could recognize the general pattern

That could work for other cases, but it doesn't work for this case:

https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs#L10-L38